src/java.base/share/classes/java/io/StreamTokenizer.java

```diff
@@ -793,30 +780,19 @@ public int lineno() {
     public String toString() {
         String ret;
         switch (ttype) {
-          case TT_EOF:
-            ret = "EOF";
-            break;
-          case TT_EOL:
-            ret = "EOL";
-            break;
-          case TT_WORD:
-            ret = sval;
-            break;
-          case TT_NUMBER:
-            ret = "n=" + nval;
-            break;
-          case TT_NOTHING:
-            ret = "NOTHING";
-            break;
-          default: {
+            case TT_EOF     -> ret = "EOF";
+            case TT_EOL     -> ret = "EOL";
+            case TT_WORD    -> ret = sval;
+            case TT_NUMBER  -> ret = "n=" + nval;
+            case TT_NOTHING -> ret = "NOTHING";
+            default         -> {
                 /*
                  * ttype is the first character of either a quoted string or
                  * is an ordinary character. ttype can definitely not be less
                  * than 0, since those are reserved values used in the previous
                  * case statements
                  */
-                if (ttype < 256 &&
-                    ((ctype[ttype] & CT_QUOTE) != 0)) {
+                if (ttype < 256 && ((ctype[ttype] & CT_QUOTE) != 0)) {
```

Comment:
Maybe (since its easier to grok the yield rather than the assignment of ret in branches):

    String ret = switch (ttype) {
            case TT_EOF     -> "EOF";
            case TT_EOL     -> "EOL";
            case TT_WORD    -> sval;
            case TT_NUMBER  -> "n=" + nval;
            case TT_NOTHING -> "NOTHING";
            default         -> {
                /*
                 * ttype is the first character of either a quoted string or
                 * is an ordinary character. ttype can definitely not be less
                 * than 0, since those are reserved values used in the previous
                 * case statements
                 */
                if (ttype < 256 && ((ctype[ttype] & CT_QUOTE) != 0)) {
                    yield sval;
                }
                char s[] = new char[3];
                s[0] = s[2] = '\'';
                s[1] = (char) ttype;
                yield new String(s);
            }
        };
        return "Token[" + ret + "], line " + LINENO;

Reply:
Code updated as suggested. See adc8af4

Reply:
The snippet above both uses yield in the default case, and also removes the assignments from the other arms. adc8af4 overlooks the redundant assignments in the non-default cases.